chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -312,7 +317,27 @@ def <=>(other)
           self.name <=> other.name
         end
 
+        # is the current disk already claimed? then use the claimed unique_name
+        def unique_name_already_claimed_by
+          claimed_disks = @node[:crowbar_wall][:claimed_disks].to_hash
+          claimed_disks.each do | claimed_name, v |
+            if self.link_to_name?(claimed_name)
+              return claimed_name
+            end
+          end
+          nil
+        end
+
         def unique_name
+          # check first if we have already a claimed disk which points to the same
+          # device node. if so, use that as "unique name"
+          already_claimed_name = self.unique_name_already_claimed_by
+          if not already_claimed_name.nil?
```

Comment:
@tboerger : Done in https://github.com/toabctl/barclamp-deployer/commit/f7bfa1bd4ddcdbc73faf7ed01016c62b906a7dc2